### DISCUSSION TITLE
I have some questions about Whisper 🙇🙇🙇

### DISCUSSION THREAD

**Discussion by NguyenMinhTri0209** [openai/whisper#1674] [Q&A]
Hi! this may be a naive question but I can't seem to figure out the answer.

Is Whisper being built based on the TensorFlow or PyTorch machine learning framework?

From what I've read, it seems that the current source code of Whisper does not contain information about how the model is built, is that correct?

Are there any other ways to fine-tune Whisper apart from using Hugging Face?

**Comment by phineas-pta** [comment]
bên huggingface ng ta làm sẵn framework tiện cho mình chạy r bác tính làm j

**Reply by NguyenMinhTri0209** [reply]
Ý mình là, thay vì load Whisper từ các model có sẵn như (tiny, base, ...) thì mình có thể tự tinh chỉnh một cái model khác (chỉ nhận dạng ngôn ngữ tiếng Việt) không ngoài dựa vào HuggingFace hay không?

**Reply by phineas-pta** [reply]
open source mà bác làm j chả dc, quan trọng là bác hiểu code ng ta ko mà tự tinh chỉnh, bác có data ko, có dàn gpu để train ko, có bik cách tính loss ko, có bik cách xử lí data ko, etc.

ng ta nguyên cả team nghiên cứu để tạo model, lấy data, xong train các kiểu, bác 1 mình 1 chợ ko bik neural network nó ra sao mà đòi tự train, bác đừng tưởng là học dc mấy khoá AI online vớ vẩn là tự train dc model

fine tune tiếng việt thì đầy ra trên huggingface, giờ bác lại đòi tự làm thì e cạn cmn lời

**Reply by NguyenMinhTri0209** [reply]
cảm ơn ông
